### DISCUSSION TITLE
Multiple Simulations and Post-Processing

### DISCUSSION THREAD

**Discussion by Swontu** [ra3xdh/qucs_s#1135] [Q&A] [answered]
I'd like to perform an AC and a NOISE simulation and compute the signal to noise ratio (SNR) at an output node in a circuit. This means I need to be able to access the results of both simulations in a post-processing step to compute it and would furthermore require an integration of the onoise_spectrum result over some bandwidth. I'd also like to be able to do parameter sweeps of component values in the circuit and plot the SNR vs. the varied parameters. I tried to use Nutmeg scripts, but failed so far. Can this be done somehow?

**Comment by tomhajjar** [comment]
Here are some projects by @dwarning that might help

[Various_projects.zip](https://github.com/user-attachments/files/18134030/Various_projects.zip)

**Reply by Swontu** [reply]
Thank you! Unfortunately, as far as I can see none of these example process both noise and ac results in a single script. Maybe the answer simply is "no" then and I need to do the post-processing externally.

**Comment by tomhajjar** [comment, marked as answer]
ngspice has always allowed as many different analysis types in the same netlist as you want. They are processed sequentially in the control section. Changes were recently made to Qucs-S to allow different analysis in the same schematic by changing the data file structure.

You can always ask questions about post processing the resultant data in the ngspice forum. They will tell you of any possible limitations. I make a schematic, generate a netlist and post the info on the ngspice forum all the time. I test the netlist using the Windows GUI, DuSpice.exe on the ngspice download page. You need to edit the netlist if you want plotting.

Posting you files speeds providing help. The ngspice of LTspice guys will always request all pertinent files. 

[Amp_100W_ngspice_prj.zip](https://github.com/user-attachments/files/18135914/Amp_100W_ngspice_prj.zip)
